api/src/main/java/jakarta/data/repository/Find.java

```diff
@@ -85,4 +125,30 @@
 @Retention(RetentionPolicy.RUNTIME)
 @Target(ElementType.METHOD)
 public @interface Find {
+    /**
+     * <p>Optionally specifies the queried entity type.</p>
+     *
+     * <p>The default value, {@code void.class}, has the special meaning of
+     * determining the entity type from the method return type if the method
+     * returns entities, and otherwise from the primary entity type of the
+     * repository.</p>
+     *
+     * <p>A repository method with the {@code Find} annotation must specify a
+     * valid entity class as the value if the method does not return entities
+     * and the repository does not otherwise define a primary entity type.</p>
+     *
+     * <p>For example,</p>
+     *
+     * <pre>
+     * &#64;Repository
+     * public interface Vehicles {
+     *     &#64;Find(Car.class)
+     *     &#64;Select("price")
```

Comment:
This goes against our spec, once "price" should not be a valid JPQL:

https://github.com/jakartaee/data/blob/main/api/src/main/java/jakarta/data/repository/Query.java#L130-L142